src/convert/convertContext.ts

```diff
@@ -49,9 +49,12 @@ export interface RecompositionState {
 class RecompositionFinalizer extends ConvertTransactionFinalizer<RecompositionState> {
   protected _state: RecompositionState = {};
 
+  // A cache of SourceComponent xml file paths to parsed contents so that identical child xml
+  // files are not read and parsed multiple times.
+  private parsedXmlCache = new Map<string, JsonMap>();
```

Comment:
It should be really small actually.  In the case of custom labels there would only be 1 entry per CustomLabels.labels-meta.xml file, which is only 1 per package directory.  This could possibly be moved to within the recompose function but I wasn't sure if there were situations where multiple parent CustomLabels would exist in a ComponentSet so to be safe I added this as an instance var.